spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/s3/S3AutoConfiguration.java

```diff
@@ -135,4 +131,36 @@ S3ObjectConverter s3ObjectConverter(Optional<ObjectMapper> objectMapper) {
 		}
 	}
 
+	@ConditionalOnMissingClass("software.amazon.awssdk.transfer.s3.S3TransferManager")
+	static class DefaultS3OutputStreamConfiguration {
+		@Bean
+		@ConditionalOnMissingBean
+		S3OutputStreamProvider diskBufferingS3StreamProvider(S3Client s3Client,
+															 Optional<S3ObjectContentTypeResolver> contentTypeResolver) {
+			return new DiskBufferingS3OutputStreamProvider(s3Client,
+				contentTypeResolver.orElseGet(PropertiesS3ObjectContentTypeResolver::new));
+		}
+	}
+
+
+	@ConditionalOnClass(S3TransferManager.class)
+	@Configuration(proxyBeanMethods = false)
+	static class S3TransferManagerConfiguration {
+		@Bean
+		S3TransferManager s3TransferManager(AwsCredentialsProvider credentialsProvider,
+				AwsRegionProvider awsRegionProvider) {
+			return S3TransferManager.builder()
+					.s3ClientConfiguration(
+							cfg -> cfg.credentialsProvider(credentialsProvider).region(awsRegionProvider.getRegion()))
```

Comment:
We should also think about adding Cross region support for `S3TransferManager`.
@maciejwalkowiak wdyt?

Reply:
Agreed with properties, also properties for things that can be set with `transferConfiguration(..)` should be exposed.

Regarding cross origin support - ideally yes, but i since i am not sure at this stage if its easily doable and generating these clients is a bit of hassle, if we decide to do it I can do it myself.

Reply:
A couple of questions to get this right.:

1. Would it be better to have a nested hierarchy under something like `spring.cloud.aws.s3.transferManager` or should I assume a flat set of properties at the current level? eg `spring.cloud.aws.s3.minimumPartSizeInBytes` vs `spring.cloud.aws.s3.transferManager.minimumPartSizeInBytes`. I tried to look up in the current example properties in this project but most of them feel very slim, although in `CredentialsProperties` the profile is considered  nested so I was planning to create the hierarchy here too
2. For naming of the properties to be exposed, I'm a fan of keeping the original property name from the library I wrap. In this case I'm asking because we have quite lengthy properties at hand (eg `minimumPartSizeInBytes` and `targetThroughputInGbps`). Shall I stick to original names or should I create contractions/slimmer names where applicable?
3. Because I need a `@ConditionalOnClass` for the new S3TransferManager, it is currently handled through a static class, but since I will need access to the `properties` bean injected in the main AutoConfiguration class, is it ok if I create an extra AutoConfiguration class and load it before the existing one? (so the one we currently have acts as default for missing Beans etc)

Sorry for the wall of text

Reply:
1. nested
2. keep original
3. 👍 

Well done @krimsz!

Reply:
Alright, should be ready to review this part. Please note that I added an extra subpackage for the property classes since felt like having inner classes would make it less readable, but I can change it if needed.
On another note, I kept a single auto-configuration test for both auto-configuration classes together because I felt it was clearer than separating it into 2, let me know the preference there